### PR TITLE
collation: add utf8_unicode_ci/utf8mb4_unicode_ci support

### PR DESCRIPTION
**Pull request by xiongjiwei**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3880
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
@wjhuang2016 Please help review this PR and confirm all applicable versions, thanks!

**Review by wjhuang2016 — COMMENTED** [review]
LGTM

File: character-set-and-collation.md [hunk 1]
```diff
@@ -453,9 +454,12 @@ ERROR 1062 (23000): Duplicate entry 'a ' for key 'PRIMARY' # TiDB 修正了 `PAD
 
 在推断排序规则时，TiDB 优先使用 coercibility 值较低的表达式的排序规则。如果 coercibility 值相同，则按以下优先级确定排序规则：
 
-binary > utf8mb4_bin > utf8mb4_general_ci > utf8_bin > utf8_general_ci > latin1_bin > ascii_bin
+binary > utf8mb4_bin > (utf8mb4_general_ci, utf8mb4_unicode_ci) > utf8_bin > (utf8_general_ci, utf8_unicode_ci) > latin1_bin > ascii_bin
```
Comment (TomShawn) [inline character-set-and-collation.md:457]: @xiongjiwei 这里表示 `utf8mb4_general_ci` 和 `utf8mb4_unicode_ci` 的优先级相同吗？

File: character-set-and-collation.md [hunk 2]
```diff
@@ -453,9 +454,12 @@ ERROR 1062 (23000): Duplicate entry 'a ' for key 'PRIMARY' # TiDB 修正了 `PAD
...
 
-如果两个子表达式的排序规则不相同，而且表达式的 coercibility 值都为 `0` 时，TiDB 无法推断排序规则并报错。
+以下情况TiDB无法推断排序规则并报错
```
Comment (TomShawn) [inline character-set-and-collation.md:459]:
```suggestion
以下情况 TiDB 无法推断排序规则并报错：
```

File: character-set-and-collation.md [hunk 3]
```diff
@@ -453,9 +454,12 @@ ERROR 1062 (23000): Duplicate entry 'a ' for key 'PRIMARY' # TiDB 修正了 `PAD
...
+
+- 如果两个子表达式的排序规则不相同，而且表达式的 coercibility 值都为 `0`。
+- 如果两个子表达式的排序规则不兼容，而且表达式的返回类型为`String`类。
```
Comment (TomShawn) [inline character-set-and-collation.md:462]:
```suggestion
- 如果两个子表达式的排序规则不兼容，而且表达式的返回类型为 `String` 类。
```

File: character-set-and-collation.md (hunk 1 above)
Comment (xiongjiwei) [reply character-set-and-collation.md:457]: 是的，相同优先级表示这些collation不兼容

Comment (TomShawn) [inline character-set-and-collation.md:457]:
```suggestion
binary > utf8mb4_bin > (utf8mb4_general_ci = utf8mb4_unicode_ci) > utf8_bin > (utf8_general_ci = utf8_unicode_ci) > latin1_bin > ascii_bin
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**